packages/integration-sdk-entity-validator/src/validator.ts

```diff
@@ -0,0 +1,69 @@
+import Ajv, { AnySchema, ErrorObject } from 'ajv';
+import addFormats from 'ajv-formats';
+import { addJ1Formats } from './j1Formats';
+import { ajvErrorToEntityValidationError } from './entityValidationError';
+
+export class EntityValidator {
+  private ajvInstance: Ajv;
+
+  constructor({ schemas }: { schemas?: AnySchema[] }) {
+    this.ajvInstance = addJ1Formats(
+      addFormats(
+        new Ajv({
+          strictSchema: false,
+          allErrors: true,
+        }),
+      ),
+    );
+
+    if (schemas) {
+      this.addSchemas(schemas);
+    }
+  }
+
+  addSchemas(schema: AnySchema | AnySchema[]) {
+    this.ajvInstance.addSchema(schema);
+    return this;
+  }
+
+  validateEntity<Entity extends { _class: string[] | string; _type: string }>(
+    entity: Entity,
+  ) {
```

Comment:
Do we want to change the parameter type to not expect an `Entity` type?

Reply:
I forgot about that. Let me fix it.